tests/test0182.d

```diff
@@ -0,0 +1,56 @@
+//T compiles:yes
+//T has-passed:yes
+//T retval:99
+
+struct S1 {
+	int a;
+	bool opEquals(S1 rhs) {
+		return a == rhs.a;
+	}
+
+	bool opEquals(const S2 rhs) {
+		return a == rhs.a;	
+	}
+
+	this(int b) {
+		a = b;
+	}
+
+}
+
+struct S2 {
+	uint a;
+	this(uint a) {
+		this.a = a;
+	}
+
+	bool opEquals(const uint a) {
+		return this.a == a;
+	}
+
+}
+
+int main() {
+
+	int a = 44;
```

Comment:
a is passed by value and, as a reuslt, never touched by the opEquals thing. The dance that is done with it in that test makes the test very hard to understand, but doesn't improve testing of opEquals in any way.

On the other, only basic opEquals behaviors are tested. Throwing some ref into the mix, having non bool return value or opEquals overrides that have side effect are thing one would expect to see in a test like this.